src/catsRouter.js

```diff
@@ -0,0 +1,164 @@
+    console.log('id being verified');
+    const id = Number(req.params.id);                   
+    if (id) {   
+        req.id = id;                                       // Alt: if (typeOf(id) === "number") { ... }        
+        console.log('id verified');
+        next() 
+    } else {
+        const newError =  new Error(`'${id}' is not a valid number`)
+        newError.status = 400;
+        next(newError); 
+    };
+}
+
+// checks object data format
+const checkObjFormat = (req, res, next) => {
+    const objectConstructor = ({}).constructor;
+    if (req.body.constructor === objectConstructor) {
+        console.log("verified request data is an object")
+        next();
+    } else {
+        const newError = new Error ('Request data is not an object - check format');
+        newError.status = 400;
+        return next(newError);
+    }
+}
+
+// checks object keys  
+const checkObjKeys = (req, res, next) => {  
+
+    const possibleKeys = ["name", "ageInYears", "favouriteToy", "description", "breedId"];
+
+    function keyNotPresent (key) { 
+        console.log(`is ${key} present?`)
+        return !possibleKeys.includes(key) // returns true if key is NOT one of the possibleKeys
+    }
+
+
+    const objToCheck = req.body;                
+    console.log(objToCheck);
+    const arrayOfKeys = Object.keys(objToCheck);      
+    console.log(arrayOfKeys);
+
+    if (arrayOfKeys.length < 1) {
+        const newError = new Error('no parameters provided');
+        newError.status = 404;
+        return next(newError);
+    }
+
+    let invalidKey = arrayOfKeys.filter(keyNotPresent); // returns an array of elements that meet the criteria 
+
+    if (invalidKey[0]) { 
+        console.log(`Property '${invalidKey[0]}' is invalid`);
+        const newError = new Error(`Property '${invalidKey[0]}' is invalid`);
+        newError.status = 400;
+        return next(newError);
+    } else {
+        console.log('object keys checked'); 
+        req.object = objToCheck; 
+        next();
+    }
+}
+
+// checks object values 
+const checkObjValues = (req, res, next) => { 
+    const objToCheck = req.body;      // { name: "", ageInYears: "", favouriteToy: ""}
+
+    for (let key in objToCheck) {
+        if (key === "ageInYears") {
+            if (isInvalidNum(objToCheck.ageInYears)) { 
+                console.log(`invalid age value: ${objToCheck.ageInYears}`);
+                return next(generateErr(objToCheck.ageInYears));
+            } 
+        } else {
+            if (isInvalidString(objToCheck[key])) { 
+                console.log(`invalid ${key} value: ${objToCheck[key]}`);
+                return next(generateErr(objToCheck[key]));
+            } 
+        };
+    }
+
+    req.object = objToCheck;            
+    console.log('object values checked');     
+    next();
+}
+
+// GET route all 
+catsRouter.get('', (req, res, next) => {
+    const cats = CatRepository.getAllCats();  //should return summary object
```

Comment:
you could consider adding a [todo/fixme](https://medium.com/imdoneio/the-imdone-pitch-feedback-welcome-386430accf01#:~:text=Most%20programmers%20put%20TODO%20comments,a%20measurement%20on%20the%20framing.&text=TODO%20comments%20are%20just%20code,%2F%2F%20TODO%20or%20%2F%2F%20FIXME.) in the comment if you want to make it easier to find :)